cloudify_azure/resources/deployment.py

```diff
@@ -88,13 +88,19 @@ def create(ctx, **kwargs):
     }
     resource_group = ResourceGroup(azure_config, ctx.logger, api_version)
     try:
-        resource_group.create_or_update(
-            resource_group_name, resource_group_params)
-    except CloudError as cr:
-        raise cfy_exc.NonRecoverableError(
-            "create deployment resource_group '{0}' "
-            "failed with this error : {1}".format(
-                resource_group_name, cr.message))
+        if resource_group.get(resource_group_name):
+            ctx.instance.runtime_properties['CREATED_RESOURCE_GROUP'] = False
```

Comment:
This should not need try/except:
```
resource_group.get(resource_group_name)
```
Why? If the resource does not exist, then the result is `None`. However, if there is some other error here, like a connection error, then that is a recoverable error and in any case a general error and does not need exception handling here.

So in that case, you can do:
```
if resource_group.get(resource_group_name):
    ctx.instance.runtime_properties['__CREATED_RESOURCE_GROUP'] = False
```
Notice that I changed `CREATED_RESOURCE_GROUP` to `__CREATED_RESOURCE_GROUP`. Internal use runtime properties usually have a `__` at the start of the string.